src/Foundation/InstalledApp.php

```diff
@@ -0,0 +1,125 @@
+<?php
+
+/*
+ * This file is part of Flarum.
+ *
+ * (c) Toby Zerner <toby.zerner@gmail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Flarum\Foundation;
+
+use Flarum\Database\Console\GenerateMigrationCommand;
+use Flarum\Database\Console\MigrateCommand;
+use Flarum\Database\Console\ResetCommand;
+use Flarum\Foundation\Console\CacheClearCommand;
+use Flarum\Foundation\Console\InfoCommand;
+use Flarum\Http\Middleware\DispatchRoute;
+use Zend\Diactoros\Response\HtmlResponse;
+use Zend\Stratigility\MiddlewarePipe;
+use function Zend\Stratigility\middleware;
+use function Zend\Stratigility\path;
+
+class InstalledApp implements AppInterface
+{
+    /**
+     * @var Application
+     */
+    protected $laravel;
+
+    /**
+     * @var array
+     */
+    protected $config;
+
+    public function __construct(Application $laravel, array $config)
+    {
+        $this->laravel = $laravel;
+        $this->config = $config;
+    }
+
+    /**
+     * @return \Zend\Stratigility\MiddlewarePipeInterface
+     */
+    public function getMiddleware()
+    {
+        if ($this->inMaintenanceMode()) {
+            return $this->getMaintenanceMiddleware();
+        } elseif ($this->needsUpdate()) {
+            return $this->getUpdaterMiddleware();
+        }
+
+        $pipe = new MiddlewarePipe;
+
+        $api = parse_url($this->laravel->url('api'), PHP_URL_PATH);
+        $pipe->pipe(path($api, $this->laravel->make('flarum.api.middleware')));
```

Comment:
this can be simplified to:

```
$pipe->pipe(path('/api', $this->laravel->make('flarum.api.middleware')));
```

getting site url and parse it to get only its path doesn't make sense.